packages/front/src/routes/dataviz/components/TableAmountsVsProgramRegion/TableAmountsVsProgramRegion.controller.ts

```diff
@@ -86,11 +86,20 @@ export class TableAmountsVsProgramRegionController {
         return Object.values(temp);
     }
 
+    private _filterYears(
+        groupedData: PartialAmountsVsProgramRegionDto[],
+        yearMin: number,
+    ): PartialAmountsVsProgramRegionDto[] {
+        const currentYear = new Date().getFullYear();
+        return groupedData.filter(row => row.exerciceBudgetaire >= yearMin && row.exerciceBudgetaire !== currentYear);
+    }
+
     public getTableData(
         data: AmountsVsProgramRegionDto[],
         selectedColumns: string[],
     ): PartialAmountsVsProgramRegionFormatted[] {
-        const temp = this._groupAndSum(data, selectedColumns);
+        let temp = this._groupAndSum(data, selectedColumns);
```

Comment:
pas fan de l'utilisation de `temp` sur différent niveau d'exécution ça rend pas clair la lecture. J'avoue que des fois c'est chiant de trouver un nom de variable, mais ça permet de comprendre le format de donnée à la lecture. Que penses-tu de `partialData` et `filteredPartialData` ou un truc comme ça ? Sinon si tu penses que c'est pas important alors on peut imbriquer comme ça :

`this._formatData(this._filterYears(this._groupAndSum(data, selectedColumns), 2021))`